### ISSUE TITLE
NER misclassified

### ISSUE THREAD

**Issue by suressha** [huseinzol05/Malaya#22]
NER  prediction is random and could not capture in case of Long Name sequences like it does not recognize middle name or sometime sirname

**Comment by huseinzol05** [comment]
Paste the code and results here like,

```python
import malaya
# start here
# output
```

I need to know which model and behavior of the output.

**Comment by huseinzol05** [comment]
I tried this news, https://www.bharian.com.my/berita/nasional/2019/08/601034/kerajaan-ada-peruntukan-laksana-program-sarapan-percuma-pm, 

```python
x = 'Kerajaan ada peruntukan untuk melaksanakan Program Sarapan Percuma (PSP) mulai tahun depan, kata Tun Dr Mahathir Mohamad. Perdana Menteri berkata, kerajaan boleh menggunakan peruntukan untuk program berkenaan daripada wang yang diperoleh daripada pelbagai agensi, misalnya Jabatan Kastam Diraja Malaysia (JKDM).'

model.predict(x)
[('Kerajaan', 'OTHER'), ('ada', 'OTHER'), ('peruntukan', 'OTHER'), ('untuk', 'OTHER'), ('melaksanakan', 'OTHER'), ('Program', 'OTHER'), ('Sarapan', 'organization'), ('Percuma', 'organization'), ('(Psp)', 'OTHER'), ('mulai', 'OTHER'), ('tahun', 'time'), ('depan', 'time'), ('kata', 'OTHER'), ('Tun', 'person'), ('Dr', 'person'), ('Mahathir', 'person'), ('Mohamad', 'person'), ('Perdana', 'person'), ('Menteri', 'person'), ('berkata', 'OTHER'), ('kerajaan', 'OTHER'), ('boleh', 'OTHER'), ('menggunakan', 'OTHER'), ('peruntukan', 'OTHER'), ('untuk', 'OTHER'), ('program', 'OTHER'), ('berkenaan', 'OTHER'), ('daripada', 'OTHER'), ('wang', 'OTHER'), ('yang', 'OTHER'), ('diperoleh', 'OTHER'), ('daripada', 'OTHER'), ('pelbagai', 'OTHER'), ('agensi', 'OTHER'), ('misalnya', 'OTHER'), ('Jabatan', 'organization'), ('Kastam', 'organization'), ('Diraja', 'organization'), ('Malaysia', 'organization'), ('(Jkdm)', 'organization')]
```

**Comment by suressha** [comment]
import malaya
malaya.entity.available_bert_model()
malaya.describe_entities()
model = malaya.entity.bert(model = 'base')
print(model.predict('Vazeer Alam Mydin Meera')) 

[('Vazeer', 'person'), ('Alam', 'location'), ('Mydin', 'organization'), ('Meera', 'organization')] 

its Name of Judge and giving it location organization as welll

**Comment by huseinzol05** [comment]
Herm, BERT model seems overfit. I tested using deep learning model `luong`,

```python
model = malaya.entity.deep_model('luong')
model.predict('Vazeer Alam Mydin Meera')
[('Vazeer', 'person'), ('Alam', 'person'), ('Mydin', 'person'), ('Meera', 'person')]
```

Malaya models trained on actual texts from news / websites, if give like literally a name, it may behave unexpectedly, I guess I need to retrain the model without need to give entire content.

Example below are better than above, string included some content `Hakim Mahkamah Tinggi`,

Luong model,
```python
model = malaya.entity.deep_model('luong')
model.predict('YA Datuk Vazeer Alam bin Mydin Meera Hakim Mahkamah Tinggi')
[('YA', 'person'), ('Datuk', 'person'), ('Vazeer', 'person'), ('Alam', 'location'), ('bin', 'person'), ('Mydin', 'organization'), ('Meera', 'person'), ('Hakim', 'person'), ('Mahkamah', 'organization'), ('Tinggi', 'organization')]
```

BERT small,
```python
model = malaya.entity.bert(model = 'small')
model.predict('YA Datuk Vazeer Alam bin Mydin Meera Hakim Mahkamah Tinggi')
[('Ya', 'person'), ('Datuk', 'person'), ('Vazeer', 'person'), ('Alam', 'person'), ('bin', 'person'), ('Mydin', 'organization'), ('Meera', 'person'), ('Hakim', 'person'), ('Mahkamah', 'organization'), ('Tinggi', 'organization')]
```

You might want to use stacking to improve the result. https://malaya.readthedocs.io/en/latest/Entities.html#voting-stack-model

**Comment by suressha** [comment]
Here i am trying to use 
import malaya

model = malaya.entity.deep_model('luong')
model.predict('Vazeer Alam Mydin Meera')

getting error when using 
  File "C:\Users\dalalns\anaconda3\lib\site-packages\malaya\texts\_text_functions.py", line 352, in generate_char_seq
    maxlen = max([j for i in x for j in i])

ValueError: max() arg is an empty sequence 
Exception: model corrupted due to some reasons, please run malaya.clear_cache('entity/luong') and try again

**Comment by suressha** [comment]
import malaya

model = malaya.entity.deep_model('luong')
model.predict('Vazeer Alam Mydin Meera')

and Adding one more problem , I am getting Random 
Exception: model corrupted due to some reasons, please run malaya.clear_cache('entity/luong') and try again

**Comment by huseinzol05** [comment]
Tried ?
```python
malaya.clear_cache('entity/luong')
```

**Comment by suressha** [comment]
Yes I did try that as it was suggested in error ,

But its not working and giving same error  message, 

Just wanted have tried this Model for BIO or BIOES tagging or do you have any BIO or Bioes tagged data for malaya NER?

**Comment by huseinzol05** [comment]
Nope, we don't have any medical data. NER in Malaya is generic.

I can't duplicate same error. Can you paste your environment here? thanks!

**Comment by huseinzol05** [comment]
```python
import malaya
model = malaya.entity.transformer(model = 'albert', size = 'base')
model.predict('Vazeer Alam Mydin Meera dan Husein makan ayam')
```

```text
[('Vazeer', 'person'),
 ('Alam', 'person'),
 ('Mydin', 'person'),
 ('Meera', 'person'),
 ('dan', 'OTHER'),
 ('Husein', 'person'),
 ('makan', 'OTHER'),
 ('ayam', 'OTHER')]
```

Fixed after released 3.0!
